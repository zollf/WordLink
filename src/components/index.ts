export { default as MenuBar } from './MenuBar';
export { default as MenuItem } from './MenuItem';
export { default as Button } from './Button';
export { default as LevelButtons } from './LevelButtons';
export { default as LevelSelect } from './LevelSelect';
export { default as Levels } from './Levels';
export { default as InputField } from './InputField';
export { default as NewUserPage } from './NewUserPage';
export { default as CircleButton } from './CircleButton';
export { default as Game } from './Game';
export { default as RegisteredUserPage } from './RegisteredUserPage';
export { default as UserStats } from './UserStats';
export { default as UserPuzzleStats } from './UserPuzzleStats';
export { default as Logo } from './Logo';
export { default as CircleText } from './CircleText';
