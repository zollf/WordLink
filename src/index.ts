export { default as Index } from './index/index';
export { default as Menu } from './menu/index';
export { default as Welcome } from './welcome/index';
export { default as Profile } from './profile/index';
export { default as Game } from './game/index';
export { default as Settings } from './settings';
