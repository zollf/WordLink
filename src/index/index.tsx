import React from 'react';
import { Text, View } from 'react-native';

import styles from './styles';

const Index = () => {
  return (
    <View style={styles.container}>
      <Text>Open up App.tsx to start working on your app!</Text>
    </View>
  );
}

export default Index;