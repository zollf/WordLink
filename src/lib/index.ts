export { default as getLevel } from './getLevel';
