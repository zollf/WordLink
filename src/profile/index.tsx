import React from 'react';
import { Text, View } from 'react-native';

import styles from './styles';

const Profile = () => {
  return (
    <View style={styles.profile}>

    </View>
  );
};

export default Profile;