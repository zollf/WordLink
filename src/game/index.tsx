import React from 'react';
import { View } from 'react-native';
import { observer } from 'mobx-react';

import { Game as WordLink } from '../../components';

import styles from './styles';

const Game = () => (
  <View style={styles.game}>
    <WordLink />
  </View>
);

export default observer(Game);