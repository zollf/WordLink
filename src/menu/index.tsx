import React from 'react';
import { View } from 'react-native';
import { observer } from 'mobx-react';

import { MenuBar } from '../../components';

import styles from './styles';

const Menu = () => {
  return (
    <View style={styles.menu}>
      <MenuBar />
    </View>
  );
};

export default observer(Menu);