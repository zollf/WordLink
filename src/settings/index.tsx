import React from 'react';
import { View } from 'react-native';
import { observer } from 'mobx-react';

import { useStore } from '../../hooks';
import { MenuBar, Button } from '../../components';

import styles from './styles';

const Settings = () => {
  const { setStartingPage, setCurrentPage } = useStore('global');

  const handleClick = () => {
    setCurrentPage('menu');
    setStartingPage(true);
  };

  return (
    <View style={styles.settings}>
      <Button text="Back" onPress={handleClick} />
      <MenuBar />
    </View>
  );
};

export default observer(Settings);
