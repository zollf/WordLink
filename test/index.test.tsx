import React from 'react';
import useWithStores from './useWithStores';

import { Index } from '../src';

import renderer from 'react-test-renderer';

const component = useWithStores(<Index />);

describe('<Index/>', () => {
  it('matches snapshot', () => {
    expect(renderer.create(component)).toMatchSnapshot();
  });
});