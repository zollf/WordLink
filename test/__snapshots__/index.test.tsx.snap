// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Index/> matches snapshot 1`] = `
<View
  style={
    Object {
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "bottom": 80,
        "left": 60,
        "position": "absolute",
        "right": 60,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#C8C8C8",
          "borderRadius": 5,
          "display": "flex",
          "flexDirection": "row",
          "height": 75,
          "justifyContent": "center",
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 2,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 4,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F3F3F3",
            "borderRadius": 5,
            "display": "flex",
            "height": 60,
            "justifyContent": "center",
            "marginLeft": 10,
            "marginRight": 10,
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
            "width": 75,
          }
        }
      >
        <Text
          style={Object {}}
        >
          Home
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F3F3F3",
            "borderRadius": 5,
            "display": "flex",
            "height": 60,
            "justifyContent": "center",
            "marginLeft": 10,
            "marginRight": 10,
            "opacity": 0.1,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
            "width": 75,
          }
        }
      >
        <Text
          style={Object {}}
        >
          Play
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F3F3F3",
            "borderRadius": 5,
            "display": "flex",
            "height": 60,
            "justifyContent": "center",
            "marginLeft": 10,
            "marginRight": 10,
            "opacity": 0.1,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
            "width": 75,
          }
        }
      >
        <Text
          style={Object {}}
        >
          Profile
        </Text>
      </View>
    </View>
  </View>
</View>
`;
